capstone/capapi/views.py

```diff
@@ -75,64 +76,38 @@ class CaseViewSet(BaseViewMixin, mixins.RetrieveModelMixin, mixins.ListModelMixi
         'court',
         'volume',
         'reporter',
-    ).prefetch_related('citations')
+    ).prefetch_related('citation')
     renderer_classes = (renderers.BrowsableAPIRenderer, renderers.JSONRenderer)
     permission_classes = (permissions.IsAPIUser,)
     filter_class = filters.CaseFilter
-    lookup_field = 'slug'
+    lookup_field = 'id'
     order_by = 'decision_date'
 
-    def download(self, **kwargs):
-        if kwargs.get(self.lookup_field):
-            try:
-                case_list = [models.CaseMetadata.objects.get(**kwargs)]
-            except models.CaseMetadata.DoesNotExist as e:
-                return JsonResponse({
-                    'message': 'Unable to find case with matching slug: %s' % e
-                }, status=404, )
-
+    def get_serializer_class(self, *args, **kwargs):
+        full_case = self.request.query_params.get('full_case', 'false').lower()
+        if full_case == 'true':
+            return serializers.CaseSerializerWithCasebody
         else:
-            cases = self.queryset.all()
-            for backend in list(self.filter_backends):
-                cases = backend().filter_queryset(self.request, cases, self)
-
-            # user is requesting a zip but there is nothing to zip, so 404 is the right response.
-            # See https://stackoverflow.com/a/11760249/307769
-            if not cases.exists():
-                return JsonResponse({
-                    'message': 'Request did not return any results.',
-                }, status=404, )
-
-            case_list = self.paginate_queryset(cases)
-
-        blacklisted_case_count = len(list((case for case in case_list if not case.jurisdiction.whitelisted)))
-
-        try:
-            # check user's case allowance against blacklisted
-            user = serializers.UserSerializer().verify_case_allowance(self.request.user, blacklisted_case_count)
-        except ValidationError as err:
-            return JsonResponse(err.detail, status=403)
-
-        filename = resources.create_zip_filename(case_list)
-
-        case_response = serializers.CaseSerializerWithCasebody(case_list, many=True, context={'request': self.request})
-        case_data = case_response.data
-        response = resources.create_download_response(filename=filename, content=case_data)
-        user.update_case_allowance(case_count=blacklisted_case_count)
-
-        return response
+            return self.serializer_class
 
     def list(self, *args, **kwargs):
-        if not self.request.query_params.get('type') == 'download':
-            return super(CaseViewSet, self).list(*args, **kwargs)
-        else:
-            return self.download(**kwargs)
+        self.serializer_class = self.get_serializer_class(self, *args, **kwargs)
```

Comment:
Don't think this line is necessary -- `self.get_serializer_class` is called automatically by `self.get_serializer`, which is called by `super`.